### PR TITLE
Conditional branching: if, '?'

### PR DESCRIPTION
**Pull request by jonnathan-ls**
# [Conditional branching: if, '?'](https://javascript.info/comparison)

## 📝 **Goal**

> Article translation

## ☑ What was done?

- [x] Translation of the article
- [x] Translation of the exercises
- [x] Translation of the SVG image

### CODE REVIEW

**Comment by jonnathan-ls** [comment]
🚧 work in progress

**Comment by jonnathan-ls** [comment]
✅ Ready for review

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 1]
```diff
@@ -124,116 +125,118 @@ if (age > 18) {
...
 
-## Non-traditional use of '?'
+## Uso não tradicional de '?'
 
-Sometimes the question mark `?` is used as a replacement for `if`:
+Às vezes, o ponto de interrogação `?` é usado como substituto para `if`:
 
 ```js run no-beautify
-let company = prompt('Which company created JavaScript?', '');
+let company = prompt('Qual empresa criou o JavaScript?', '');
```
Comment (odsantos) [inline 1-js/02-first-steps/10-ifelse/article.md:210]:
```suggestion
let company = prompt('Que empresa criou o JavaScript?', '');
```

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 2]
```diff
@@ -124,116 +125,118 @@ if (age > 18) {
...
+Dependendo da condição `company == 'Netscape'`, a primeira ou a segunda expressão após o `?` é executada e mostra um alerta.
 
-We don't assign a result to a variable here. Instead, we execute different code depending on the condition.
+Não atribuímos um resultado a uma variável aqui. Em vez disso, executamos um código diferente dependendo da condição.
 
-**It's not recommended to use the question mark operator in this way.**
+**Não é recomendado usar o operador de ponto de interrogação dessa maneira.**
 
-The notation is shorter than the equivalent `if` statement, which appeals to some programmers. But it is less readable.
+A notação é mais curta do que a instrução `if` equivalente, que atrai alguns programadores. Mas é menos legível.
```
Comment (odsantos) [inline 1-js/02-first-steps/10-ifelse/article.md:224]:
```suggestion
A notação é mais curta do que a instrução `if` equivalente, o que atrai alguns programadores. Mas é menos legível.
```

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 3]
```diff
@@ -124,116 +125,118 @@ if (age > 18) {
...
 
-Here is the same code using `if` for comparison:
+Aqui está o mesmo código usando `if` para comparação:
 
 ```js run no-beautify
-let company = prompt('Which company created JavaScript?', '');
+let company = prompt('Qual empresa criou o JavaScript?', '');
```
Comment (odsantos) [inline 1-js/02-first-steps/10-ifelse/article.md:229]:
```suggestion
let company = prompt('Que empresa criou o JavaScript?', '');
```

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 4]
```diff
@@ -124,116 +125,118 @@ if (age > 18) {
...
+  alert('Errado.');
 }
 */!*
 ```
 
-Our eyes scan the code vertically. Code blocks which span several lines are easier to understand than a long, horizontal instruction set.
+Nossos olhos escaneiam o código verticalmente. Blocos de código que abrangem várias linhas são mais fáceis de entender do que um conjunto de instruções longo e horizontal.
 
-The purpose of the question mark operator `?` is to return one value or another depending on its condition. Please use it for exactly that. Use `if` when you need to execute different branches of code.
+O objetivo do operador de ponto de interrogação `?` é retornar um valor ou outro dependendo de sua condição. Por favor, use-o exatamente para isso. Use `if` quando precisar executar diferentes ramificações de código.
```
Comment (odsantos) [inline 1-js/02-first-steps/10-ifelse/article.md:242]:
```suggestion
O objetivo do operador de ponto de interrogação `?` é retornar um valor ou outro dependendo de sua condição. Por favor, use-o exatamente para isso. Use `if` quando precisar de executar diferentes ramificações de código.
```

**Comment by odsantos** [comment]
Hi @jonnathan-ls ,

I updated to the English version.
As part of the pending tasks could you also resolve the conflicts?

Thanks, 
Osvaldo.

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 5]
```diff
@@ -1,117 +1,118 @@
-# Conditional branching: if, '?'
+# Ramificação condicional: if, '?'
 
-Sometimes, we need to perform different actions based on different conditions.
+Às vezes, precisamos realizar ações diferentes com base em condições diferentes.
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:3]: Não acho que a preposição  `de`  se encaixa corretamente entre a expressão `precisamos realizar`

Avaliando a tradução literal:

image

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 6]
```diff
@@ -1,117 +1,118 @@
...
 
-The `if(...)` statement evaluates a condition in parentheses and, if the result is `true`, executes a block of code.
+A instrução `if(...)` avalia uma condição entre parênteses e, se o resultado for `true`, executa um bloco de código.
 
-For example:
+Por exemplo:
 
 ```js run
-let year = prompt('In which year was ECMAScript-2015 specification published?', '');
+let year = prompt('Em que ano a especificação ECMAScript-2015 foi publicada?', '');
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:14]: Fica um estranho a leitura para `foi a especificação` 

Mas revendo o que escrevi e considerando a tradução literal de

`In which year was ECMAScript-2015 specification published?`

Alterei para:

`Em que ano foi publicada a especificação ECMAScript-2015?`

Me parece mais sucinto e direto para a interpretação do leitor.

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 7]
```diff
@@ -1,117 +1,118 @@
...
 
-The `if` statement may contain an optional "else" block. It executes when the condition is falsy.
+A instrução `if` pode conter um bloco opcional "else". Ele é executado quando a condição é falsa.
+
+Por exemplo:
 
-For example:
 ```js run
-let year = prompt('In which year was the ECMAScript-2015 specification published?', '');
+let year = prompt('Em que ano a especificação ECMAScript-2015 foi publicada?', '');
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:76]: Vide comentário https://github.com/javascript-tutorial/pt.javascript.info/pull/246/files#r1148668485

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 8]
```diff
@@ -1,117 +1,118 @@
...
 
-Sometimes, we'd like to test several variants of a condition. The `else if` clause lets us do that.
+Às vezes, gostaríamos de testar muitas variantes de uma condição. A cláusula `else if` nos permite fazer isso.
 
-For example:
+Por exemplo:
 
 ```js run
-let year = prompt('In which year was the ECMAScript-2015 specification published?', '');
+let year = prompt('Em que ano a especificação ECMAScript-2015 foi publicada?', '');
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:92]: Vide comentário https://github.com/javascript-tutorial/pt.javascript.info/pull/246/files#r1148668485

File: 1-js/02-first-steps/10-ifelse/article.md [hunk 9]
```diff
@@ -1,117 +1,118 @@
...
+No código acima, o JavaScript primeiro verifica `year < 2015`. Se for falso, vai para a próxima condição `ano > 2015`. Se isso também for falso, mostra o último 'alerta'.
 
-There can be more `else if` blocks. The final `else` is optional.
+Pode haver mais blocos `else if`. O `else` final é opcional.
 
-## Conditional operator '?'
+## Operador condicional '?'
 
-Sometimes, we need to assign a variable depending on a condition.
+Às vezes, precisamos atribuir uma variável dependendo de uma condição.
```
[the remark this answers is not in the document]
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:109]: Semelhante a https://github.com/javascript-tutorial/pt.javascript.info/pull/246/files#r1148666022

O que acha de ?

> Às vezes, é preciso atribuir uma variável dependendo de determinada condição.

File: 1-js/02-first-steps/10-ifelse/article.md (hunk 4 above)
Comment (jonnathan-ls) [reply 1-js/02-first-steps/10-ifelse/article.md:242]: Semelhante a https://github.com/javascript-tutorial/pt.javascript.info/pull/246/files#r1148666022

Sem a preposição fica o mais próximo da tradução literal: 

image

**Comment by jonnathan-ls** [comment]
> Hi @jonnathan-ls ,
> 
> I updated to the English version. As part of the pending tasks could you also resolve the conflicts?
> 
> Thanks, Osvaldo.

@odsantos conflitos resolvidos.

**Comment by jonnathan-ls** [comment]
> Hi, Please fell free to tell where you disagree. Thanks, Osvaldo.

@odsantos 
Correções realizadas.

Apenas algumas sugestões relacionadas a preposição "de" que não fica um pouco estranho para o leitor.

- https://github.com/javascript-tutorial/pt.javascript.info/pull/246#discussion_r1148672070
- https://github.com/javascript-tutorial/pt.javascript.info/pull/246#discussion_r1148666022
- https://github.com/javascript-tutorial/pt.javascript.info/pull/246#discussion_r1148669533

**Comment by nazarepiedady** [comment]
@jonnathan-ls have you time to update this translation and correct its conflicts?

**Comment by jonnathan-ls** [comment]
> @jonnathan-ls have you time to update this translation and correct its conflicts?

Sure, I'll do it on the weekend

**Comment by nazarepiedady** [comment]
@jonnathan-ls I also recommend looking at all your pull requests to see what more needs to be done to let the pull request fine be applied.

**Comment by jonnathan-ls** [comment]
Conflicts resolved @nazarepiedady 

@odsantos awaiting analysis of the corrections I made according to your review

**Comment by nazarepiedady** [comment]
@jonnathan-ls Are all reviews requested by @odsantos resolved?

**Comment by jonnathan-ls** [comment]
> Are all reviews requested by @odsantos resolved?

Yes, i am waiting his to review again

**Pull request merged**